app/views/reports/index.html.erb

```diff
@@ -221,3 +221,30 @@
     </div>
   </div>
 </div>
+
+<div class="modal fade" id="formatSelectionModal" tabindex="-1" role="dialog" aria-labelledby="formatSelectionModalLabel" aria-hidden="true">
+  <div class="modal-dialog  modal-dialog-centered" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="formatSelectionModalLabel">Select Format</h5>
+        <button type="button" class="close" data-bs-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <p>Select the export format:</p>
+        <button type="button" class="btn btn-primary" onclick="updateFormat('csv')" data-bs-dismiss="modal">CSV </button>
+        <button type="button" class="btn btn-primary" onclick="updateFormat('xlsx')" data-bs-dismiss="modal">Excel</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
```

Comment:
Not a fan of putting JS in a view. This should normally be separated into a JS file, but you can achieve the same result by moving the modal inside the form and letting the CSV and Excel buttons be your submit buttons. You just need to put  name and value tags on the buttons.